app/controllers/checkout_controller.rb

```diff
@@ -201,4 +204,29 @@ def redirect_to_paypal_express_form_if_needed
     render json: {path: spree.paypal_express_url(payment_method_id: payment_method.id)}, status: 200
     true
   end
+
+  def construct_saved_card_attributes
+    existing_card_id = params[:order].delete(:existing_card)
```

Comment:
Resolved

Reply:
For reference this was written the way it was because `existing_card` is the key used by [future versions of Spree](https://github.com/spree/spree/blob/2-3-stable/core/app/models/spree/order/checkout.rb#L234).